packages/backend/src/backend_factory.ts

```diff
@@ -108,6 +120,51 @@ export class BackendFactory<
   createStack = (name: string): Stack => {
     return this.stackResolver.createCustomStack(name);
   };
+
+  setCustomOutput = (
+    key: string,
+    value: string,
+    options?: CustomOutputOptions
+  ): void => {
+    /*
+    Looks like this must be in certain format.
+    From attempt to use non-alpha numeric characters with this.outputStorageStrategy.addBackendOutputEntry() in early prototype.
+
+    The CloudFormation deployment has failed. Find more information in the CloudFormation AWS Console for this stack.
+    Caused By: ❌ Deployment failed: Error [ValidationError]: Template format error: Outputs name 'amplify-custom-myApiUrl-value' is non alphanumeric.
+     */
+
+    if (!this.customOutputsEntry) {
+      this.customOutputsEntry =
+        this.outputStorageStrategy.addAppendableBackendOutputEntry(
+          'AWS::Amplify::Custom'
+        );
+    }
+
+    const outputData = JSON.stringify({ value: value, options });
+    console.log(outputData);
+    // TODO how do we do custom namespacing ??
+    this.customOutputsEntry.appendOutput(`amplifycustom${key}`, outputData);
+  };
+
+  addToClientConfig = (clientConfigPart: Partial<ClientConfig>) => {
+    // Lets see if this can work.
+    // This implementation doesn't do much, it just tries to see if we can serialize cdk tokens.
+
+    if (!this.customOutputsEntryAlternative) {
+      this.customOutputsEntryAlternative =
+        this.outputStorageStrategy.addAppendableBackendOutputEntry(
+          'AWS::Amplify::Custom::Alternative'
+        );
+    }
+
+    const outputData = JSON.stringify(clientConfigPart);
+    this.customOutputsEntryAlternative.appendOutput(
+      `alternativeCustomApproach${this.customOutputsAlternativeEntryCount}`,
+      outputData
+    );
+    this.customOutputsAlternativeEntryCount++;
+  };
 }
```

Comment:
Added a sample of how this could work.

A sample output looks like this:
```
{"myApiUrl":"https://REDACTED.execute-api.us-west-2.amazonaws.com/prod/","myApiKey":"REDACTED"}}
```